src/swarmManager.test.js

```diff
@@ -0,0 +1,59 @@
+const {swarmManager} = require('./swarmManager');
+const {invoke} = require('lodash/fp');
+
+
+require('../tests/test.configurations');
+
+describe('swarmManager', function () {
+
+    const numberOfDaemons = 3;
+
+    beforeEach('generateSwarms', async function () {
+        this.timeout(10000);
+
+        this.swarmManager = await swarmManager();
+        this.swarm0 = await this.swarmManager.generateSwarm({numberOfDaemons});
+        this.swarm1 = await this.swarmManager.generateSwarm({numberOfDaemons});
+    });
+
+    afterEach('stop swarm', async function () {
+        await this.swarmManager.stopAll();
+        this.swarmManager.removeSwarmState();
+    });
+
+    it('should be able to start all nodes', async function () {
+        requestIsRunningStatusFromAllNodes.call(this)
+            .should.all.be.eq(false);
+
+        await this.swarmManager.startAll();
+
+        requestIsRunningStatusFromAllNodes.call(this)
+            .should.all.be.eq(true);
+    });
+
+    it('should be able to stop all nodes', async function () {
+        await this.swarmManager.startAll();
+
+        requestIsRunningStatusFromAllNodes.call(this)
+            .should.all.be.eq(true);
+
+        await this.swarmManager.stopAll();
+
+        requestIsRunningStatusFromAllNodes.call(this)
+            .should.all.be.eq(false);
+    });
+
+    it('swarms should have the same api', function () {
+        JSON.stringify(this.swarm0).should.deep.equal(JSON.stringify(this.swarm1));
+    });
+
+    it('swarms should have different peers lists', function () {
+        this.swarm0.getPeersList().should.not.deep.equal(this.swarm1.getPeersList());
+    });
+});
+
+function flattenArray(acc, curr) { return acc = [...acc, ...curr] };
```

Comment:
Nit: This isn't flattening, but joining.

Reply:
```
const nestedArray = [ [ 1, 2, 3 ], [ 'a', 'b', 'c' ] ];

// I only needed to flatten one level deep, can you suggest a better name?
nestedArray.reduce((acc, arr) => [...acc, ...arr], []);
> [ 1, 2, 3, 'a', 'b', 'c' ]

// I read that using spread operator in a reduce is kind of an anti pattern. would you say this is preferable, albeit slightly more verbose? 
nestedArray.reduce((acc, arr) => {arr.forEach(el => acc.push(el)); return acc}, [])
> [ 1, 2, 3, 'a', 'b', 'c' ]

```

Reply:
If `nestedArray` was passed to your function as the only arg, then this would indeed be a "flatten" operation. I actually missed how you were calling this though, which indeed is in a reduce. I'd expect there may be some JS naming convention to clarify that something is a reducer function? (If there is, I am not familiar with it).

I'm not against using the spread because it's perfectly clear IMO (and shorter, bonus), but if there's any existing style guide you normally conform to I would go with its recommendation.